### PR TITLE
implement skip row option

### LINKED ISSUE #3869: Feature: support SKIP rows in LOAD FROM for COPY
### API

C++

### Description

Sometimes CSV can have multiple lines of headers. It can be helpful to skip lines while copying from CSVs semantically similar to `COPY ... FROM (LOAD FROM 'csv' RETURN * SKIP ...) `

### PR DESCRIPTION
**Pull request by acquamarin**
# Description

This PR adds skip row option to the CSV reader.

Fixes #3869

### CODE REVIEW

File: src/include/common/constants.h [hunk 1]
```diff
@@ -143,6 +143,9 @@ struct CopyConstants {
     static constexpr char DEFAULT_CSV_LINE_BREAK = '\n';
     static constexpr const char* ROW_IDX_COLUMN_NAME = "ROW_IDX";
     static constexpr uint64_t PANDAS_PARTITION_COUNT = 50 * DEFAULT_VECTOR_CAPACITY;
+
+    static constexpr const char* INT_CSV_PARSING_OPTIONS[] = {"SKIP"};
```
Comment (andyfengHKU) [inline src/include/common/constants.h:147]: if this is csv specific. Let's move this to csv_reader_config.h

File: test/test_files/copy/copy_with_skip_lines.test [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+-STATEMENT LOAD FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' (skip=2, HEADER=TRUE) RETURN *;
+---- 3
+3|Carol|1|false|true|45|5.0|1940-06-22|1911-08-20 02:32:21|48 hours 24 minutes 11 seconds|[4,5]|[Carmen,Fred]|[[8,10]]|[91,75,21,95]|1.00|a0eebc999c0b4ef8bb6d6bb9bd380a13
+5|Dan|2|false|true|20|4.8|1950-7-23|2031-11-30 12:25:30Z|10 years 5 months 13 hours 24 us|[1,9]|[Wolfeschlegelstein,Daniel]|[[7,4],[8,8],[9]]|[76,88,99,89]|1.30|a0ee-bc99-9c0b-4ef8-bb6d-6bb9-bd38-0a14
+7|Elizabeth|1|false|true|20|4.7|1980-10-26|1976-12-23 11:21:42|48 hours 24 minutes 11 seconds|[2]|[Ein]|[[6],[7],[8]]|[96,59,65,88]|1.463|{a0eebc99-9c0b4ef8-bb6d6bb9-bd380a15}
+
+-CASE CopyFromWithSkippedLines
+-STATEMENT create node table person (ID INt64, fName StRING, gender INT64, isStudent BoOLEAN, isWorker BOOLEAN, age INT64, eyeSight DOUBLE, birthdate DATE, registerTime TIMESTAMP, lastJobDuration interval, workedHours INT64[], usedNames STRING[], courseScoresPerTerm INT64[][], grades INT64[4], height float, u UUID, PRIMARY KEY (ID));
+---- ok
+-STATEMENT COPY person FROM '${KUZU_ROOT_DIRECTORY}/dataset/tinysnb/vPerson.csv' (skip=3, HEADER=TRUE);
```
Comment (andyfengHKU) [inline test/test_files/copy/copy_with_skip_lines.test:15]: I want to check several things

- If this is called `offset` in DuckDB? Let's try to be consistent if there is an example in the market already.
- Does it make sense to make it work for parquet or pandas?

**Review by andyfengHKU — APPROVED** [review]

File: src/include/common/constants.h (hunk 1 above)
Comment (acquamarin) [reply src/include/common/constants.h:147]: I think we keep all copyConstants in the `constants.h` rather `than csv_reader_config.h`

File: test/test_files/copy/copy_with_skip_lines.test (hunk 2 above)
Comment (acquamarin) [reply test/test_files/copy/copy_with_skip_lines.test:15]: https://duckdb.org/docs/data/csv/overview.html
Duckdb also uses the term `skip` in csv reader.
We should also delegate this option to parquet and pandas but let's leave it to next release.

**Pull request merged**